### PR TITLE
(feat): Assessment report tool

### PR DESCRIPTION
**Pull request by TylerMizuyabu**
Introduces a command to the gh_foundations cli tool to run checks against your github organization and repositories then output a report in a json format. The command requires either `GITHUB_TOKEN` environment variable to be set or gh cli to be installed and for the user to be logged in to authenticate against githubs REST api. For it's first iteration there is only one set of checks performed against the Gov of Canada's gaurdrails. More check types can be added later along with a command line option to configure what subset of checks get run or if all of them get run.

### CODE REVIEW

**Comment by blastdan** [comment]
Do you know if there is a way to use default credentials using the GH CLI that is easy? Just thinking about keeping the auth methods for everything the same.

File: organizations/tools/gh_foundations/internal/pkg/types/github/organization.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+			report.Results[t] = r
+		default:
+			report.Results[t] = types.NotApplicable
+		}
+	}
+	return report
+}
+
+// Custom repository roles for an organization need to be accessed separately from settings
+
```
Comment (bzarboni1) [inline organizations/tools/gh_foundations/internal/pkg/types/github/organization.go:39]:
```suggestion
```

File: organizations/tools/gh_foundations/internal/pkg/types/github/repository.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+			report.Results[t] = types.NotApplicable
+		}
+	}
+	return report
+}
+
+func (r *Repository) GoCGaurdrailsCompliant() (types.CheckResult, *types.CheckError) {
+	var allErrors error
+	violations := make(map[string]string)
+	checks := []func(repo *Repository) (string, error){
```
Comment (bzarboni1) [inline organizations/tools/gh_foundations/internal/pkg/types/github/repository.go:43]:
```suggestion
	checks := []func(repo *Repository) (string, error) {
```

**Review by bzarboni1 — APPROVED** [review]

**Comment by TylerMizuyabu** [comment]
> Do you know if there is a way to use default credentials using the GH CLI that is easy? Just thinking about keeping the auth methods for everything the same.

After looking through the gh cli docs the best I think we could do would be to change the env var we use to be `GH_TOKEN` to match the variable used by github. I'm not seeing anything from them where they setup default credentials like gcloud does.

**Review by bzarboni1 — APPROVED** [review]

**Review by bzarboni1 — APPROVED** [review]

**Review by bzarboni1 — APPROVED** [review]

**Pull request merged**